modal/_partial_function.py

```diff
@@ -922,3 +922,174 @@ def wrapper(
         return pf
 
     return wrapper
+
+
+# NOTE: torchrun is currently exposed through modal.experimental, not the top-level namespace
+def _torchrun(
+    _warn_parentheses_missing=None,
+    *,
+    nproc_per_node: int = 1,
+    master_port: int = 29500,
+) -> Callable[
+    [Union[_PartialFunction[P, ReturnType, ReturnType], Callable[P, ReturnType]]],
+    _PartialFunction[P, ReturnType, ReturnType],
+]:
+    """Decorator that enables torchrun orchestration for distributed training.
+
+    When used with @modal.experimental.clustered, this decorator automatically handles
+    the torchrun orchestration, allowing you to write your training logic directly
+    in the decorated function without needing a separate script file.
+
+    The function will be called in two modes:
+    - Orchestration mode: Sets up and launches torchrun
+    - Data mode: Executes the actual training logic (called by torchrun)
+
+    Parameters:
+    nproc_per_node: int = 1
+        Number of processes to launch per node
+    master_port: int = 29500
+        Port for inter-process communication
+
+    Example:
+    ```python
+    @app.function(gpu="H100:8")
+    @modal.experimental.clustered(4, rdma=True)
+    @modal.experimental.torchrun(nproc_per_node=8)
```

Comment:
The `nproc_per_node` is redundant with the GPU specification in the function decorator right? I wonder if we could implicitly make that connection behind the scenes.